packages/contracts-bedrock/CONTRIBUTING.md

```diff
@@ -0,0 +1,118 @@
+- **Explain why this enhancement would be useful** in Optimism's smart contracts. You may also want to point out the other projects that solved it better and which could serve as inspiration.
+
+### Your First Code Contribution
+
+The best place to begin contributing is by looking through the issues with the `good first issue` label. These are issues that are relatively easy to implement and are a great way to get familiar with the codebase.
+
+Optimism's smart contracts are written in Solidity and we use [foundry](https://github.com/foundry-rs/foundry) as our development framework. To get started, you'll need to install several dependencies:
+1. [pnpm](https://pnpm.io)
+  1. Make sure to `pnpm install`
+1. [foundry](https://getfoundry.sh)
+  1. Foundry is built with [rust](https://www.rust-lang.org/tools/install), and this project uses a pinned version of foundry. Install the rust toolchain with `rustup`.
+  1. Make sure to install the version of foundry used by `ci-builder`, defined in the `.foundryrc` file in the root of this repo. Once you have `foundryup` installed, there is a helper to do this: `pnpm install:foundry`
+1. [golang](https://golang.org/doc/install)
+1. [python](https://www.python.org/downloads/)
+
+Our [Style Guide](STYLE_GUIDE.md) contains information about the project structure, syntax preferences, naming conventions, and more. Please take a look at it before submitting a PR, and let us know if you spot inconcistencies!
+
+Once you've read the styleguide and are ready to work on your PR, there are a plethora of useful `pnpm` scripts to know about that will help you with development:
+1. `pnpm build` Builds the smart contracts.
+1. `pnpm test` Runs the full `forge` test suite.
+1  `pnpm gas-snapshot` Generates the gas snapshot for the smart contracts.
+1. `pnpm semver-lock` Generates the semver lockfile.
+1. `pnpm storage-snapshot` Generates the storage lockfile.
+1. `pnpm autogen:invariant-docs` Generates the invariant test documentation.
+1. `pnpm clean` Removes all build artifacts for `forge` and `go` compilations.
+1. `pnpm validate-spacers` Validates the positions of the storage slot spacers.
+1. `pnpm validate-deploy-configs` Validates the deployment configurations in `deploy-config`
+1. `pnpm slither` Runs the slither static analysis tool on the smart contracts.
+1. `pnpm lint` Runs the linter on the smart contracts and scripts.
+1. `pnpm pre-pr` Runs most checks, generators, and linters prior to a PR. For most PRs, this is sufficient to pass CI if everything is in order.
+1. `pnpm pre-pr:full` Runs all checks, generators, and linters prior to a PR.
+
+### Improving The Documentation
+
+Documentation improvements are more than welcome! If you see a typo or feel that a code comment describes something poorly or incorrectly, please submit a PR with a fix.
+
+### Deploying on Devnet
+
+To deploy the smart contracts on a local devnet, run `make devnet-up` in the monorepo root.
```

Comment:
```suggestion
To deploy the smart contracts on a local devnet, run `make devnet-up` in the monorepo root. This will serve an rpc endpoint at localhost on port `8545` (qualified as: `127.0.0.1:8545`). When using `forge`, this is the default rpc url, so you shouldn't need to pass it in with the `--rpc-url 127.0.0.1:8545` flag.
```

Reply:
hmm I think this isn't accurate wrt the rpc url flag. Maybe we point to the devnet documentation? Gives more info than a sentence or two could.

Reply:
Talked offline, added a link to `devnet.md`